--- libarchive/archive_read_disk_posix-org.c	2014-12-14 11:16:36.446209080 -0500
+++ libarchive/archive_read_disk_posix.c	2014-12-14 11:18:27.542209065 -0500
@@ -93,6 +93,12 @@ __FBSDID("$FreeBSD$");
 #include <sys/ioctl.h>
 #endif
 
+#ifdef __ANDROID__
+#include <sys/vfs.h>
+#define statvfs statfs
+#define fstatvfs fstatfs
+#endif
+
 #include "archive.h"
 #include "archive_string.h"
 #include "archive_entry.h"

